contracts/on-chain-claim/src/contract.rs

```diff
@@ -119,9 +123,32 @@ pub trait OnChainClaimContract:
         );
     }
 
+    #[endpoint(addSeason)]
+    fn add_season(&self, epoch: u64) {
+        self.require_caller_is_admin();
+
+        if self.seasons().is_empty() {
+            let mut seasons = ManagedVec::new();
+            seasons.push(SeasonInfo::new(FIRST_SEASON_ID, epoch));
+            self.seasons().set(seasons);
```

Comment:
If we use the new approach with the VecMapper, I'd add a new check. The last season must be the current season. 
So seasons().get(seasons().len) < current_epoch. This way, you'll always have the current season either the last element or the one before (count() or count() - 1).